android_vs_ios.md

```diff
@@ -0,0 +1,175 @@
+This document tries to provide a high level overview of some key differences (and similarities) of developing for Android versus developing for iOS. It is aimed to be a reference for people who are starting to or are considering switching from one mobile platform to another.
+
+# Language
+
+**TL;DR: Kotlin is more convenient, Swift is more powerful**
+
+## Common features and similarities (Kotlin and Swift)
+* Mainly object oriented (classes, inheritance, interface/protocol)
+* Statically typed
+* Type inference with option to specify explicit type
+* Type alias
+* Extensions
+* Generics
+* Optionals
+* Properties
+* Higher order functions
+* Default method parameters
+* Operator overloading
+* Interoperable with the previous language (with Java and Objective C respectively)
+* Semicolons are optional
+
+## Memory management
+Kotlin and Java uses [garbage collection](https://en.wikipedia.org/wiki/Garbage_collection_(computer_science)), which handles pretty much all the memory deallocation automatically, with the downside being that it can cause performance issues (ex: frame drop) when it runs and it is very hard/impossible to predict when it is triggered. Memory leaks can still happen (especially on Android), but there's not much concern for cyclic references.
```

Comment:
What do you mean by `there's not much concern for cyclic references`? Do you mean they're unlikely to happen, or that nobody cares when they *do* happen? I haven't done much Android or (most of my Java was Swing), but we used to get cyclic references and memory leaks all the time.